payment/models.py

```diff
@@ -58,6 +60,53 @@ class PaymentRequest(ModelWithStr):
     callback = StringType()  # a webhook to call when a payment is complete
 
 
+class WhitelistRequest(ModelWithStr):
+    order_id = StringType()
+    source = StringType()
+    destination = StringType()
+    amount = IntType()
+    xdr = StringType()
+    network_id = StringType()
+    app_id = StringType()
+
+    @staticmethod
+    def compare_attr(attr1, attr2, attr_name):
+        if attr1 != attr2:
+            raise TransactionMismatch('{attr_name}: {attr1} does not match expected {attr_name}: {attr2}'.
+                                      format(attr_name=attr_name,
+                                             attr1=attr1,
+                                             attr2=attr2))
+
+    def verify_transaction(self):
+        """Verify that the encoded transaction matches our expectations"""
+        try:
+            decoded_tx = decode_transaction(self.xdr, self.network_id)
```

Comment:
why is network_id given? isn't it always the same?

Reply:
network_id is needed to calculate the tx hash and for signing it. When we built the whitelisting we decided it was safer to verify that the client and server are on the same network than just assume it

Reply:
What types of networks do you have? Is it production vs playground or versions 2 vs 3?

In any case if the client claims to be on a specific network but it actually is on a different one, you won't have anything to do here.
I feel it's weird to check for a value that should be hardcoded in the deployment level.

Reply:
Network id is the network passphrase ('Kin Playground Network ; June 2018')
I agree that its a bit strange here, we will probably need to change it later in the sdk level

Reply:
agree with Doody, shouldn't be verified in this level